web-portal/backend/src/authkeys/authkeys.controller.ts

```diff
@@ -0,0 +1,14 @@
+import { Controller, Post, Param } from '@nestjs/common';
+import { AuthkeysService } from './authkeys.service';
+
+@Controller('tenant')
+export class AuthkeysController {
+  constructor(private readonly authKeyService: AuthkeysService) {}
+
+  @Post(':tenantId/authkey')
+  async applyCredits(@Param('tenantId') tenantId: string): Promise<any> {
```

Comment:
This function name is confusing, should be `createAuthKey()`?

Reply:
Also, this should require the secretkey as a parameter to authorize this action. The confusion between the tenantId and the secret key should be rectified as I was stuck on the secret being the primary key, but either way it needs to be clear where the authentication/authorization lies.